core/src/main/java/org/infinispan/stream/StreamMarshalling.java

```diff
@@ -138,7 +169,7 @@ public Integer getId() {
       public void writeObject(ObjectOutput output, Object object) throws IOException {
          int number = objects.get(object.getClass(), -1);
          output.writeByte(number);
-         switch (number) {
+         switch (ExternalizerId.values()[number]) {
```

Comment:
If you put the enums in the map instead of their ordinals, I think you can skip the `values()` access. But you should probably add a null check here as well.